src/backend/parser/parse_agg.c

```diff
@@ -840,35 +847,47 @@ parseCheckAggregates(ParseState *pstate, Query *qry)
 	}
 
 	/*
-	 * Build a list of the acceptable GROUP BY expressions for use by
+	 * Build a list of the acceptable unique GROUP BY expressions for use by
 	 * check_ungrouped_columns().
+	 *
```

Comment:
Can't we replace the entire loop below with a walker similar to [CQueryMutators::GetGroupUniqueTleReferencesWalker](https://github.com/arenadata/gpdb/blob/ae07e294ffce0edd763529acc0a9aac4ec7f1f8b/src/backend/gpopt/translate/CQueryMutators.cpp#L179)

Reply:
Yes, this is very similar to what we need. But in this cycle we need to get TLE, not refsortgroupref. Because in further processing either TLE or expr is used.